### PR TITLE
Move initialization of vector and preconditioner into their own function

### PR DESCRIPTION
**Pull request by BenjaminRodenberg**
## Main changes of this PR

I split the initialization procedure into smaller functions following the suggestions in #2005.

## Motivation and additional information

Refactoring as a preparation for #2005.

## Author's checklist

* [x] I used the [`pre-commit` hook](https://precice.org/dev-docs-dev-tooling.html#setting-up-pre-commit) to prevent dirty commits and used `pre-commit run --all` to format old commits.
* [x] I added a changelog file with `make changelog` if there are user-observable changes since the last release.
* [x] I added a test to cover the proposed changes in our test suite.
* [x] For breaking changes: I documented the changes in the appropriate [porting guide](https://precice.org/couple-your-code-porting-overview.html).
* [x] I stuck to C++17 features.
* [x] I stuck to CMake version 3.22.1.
* [x] I squashed / am about to squash all commits that should be seen as one.

## Reviewers' checklist

* [ ] Does the changelog entry make sense? Is it formatted correctly?
* [ ] Do you understand the code changes?

### CODE REVIEW

File: src/acceleration/IQNIMVJAcceleration.cpp [hunk 1]
```diff
@@ -74,50 +74,11 @@ void IQNIMVJAcceleration::initialize(
 {
   PRECICE_TRACE();
 
-  // do common QN acceleration initialization
-  BaseQNAcceleration::initialize(cplData);
-
   if (_imvjRestartType > 0)
     _imvjRestart = true;
```
[the remark this answers is not in the document]
Comment (fsimonis) [reply src/acceleration/IQNIMVJAcceleration.cpp:78]: `_imvjRestart` depends on `_imvjRestartType`, which is only set in the constructor.

Thus, you can move this into the constructor and simplify thing further.
At that point, the complete initialize could be removed from the IQNIMVJAcceleration.

**Review by fsimonis — APPROVED** [review]
Sane refactoring with a minor hint.

File: src/acceleration/IQNIMVJAcceleration.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (BenjaminRodenberg) [reply src/acceleration/IQNIMVJAcceleration.cpp:78]: I was intentionally ignoring this :smirk: But the suggestion, of course makes sense. If I'm already touching the class and we start discussing it, I can also move it. But let's try to not refactor the whole class please :pray:

[the remark this answers is not in the document]
Comment (BenjaminRodenberg) [reply src/acceleration/IQNIMVJAcceleration.cpp]: Good point. Now we do not need to override `initialization()` in the subclasses anymore and `BaseQNAcceleration::initialization()` becomes `final`. :+1: 

Added changes in 2777c661a2e8a6df30bd24fb10290f0457c09807.

**Comment by BenjaminRodenberg** [comment]
@NiklasKotarsky do you want to review this PR? It is directly connected to #2005 and more meant as an fyi for you. After merging this PR into develop, I can also take care of synchronizing #2005 to the updated `develop`.

**Comment by NiklasKotarsky** [comment]
> @NiklasKotarsky do you want to review this PR? It is directly connected to #2005 and more meant as an fyi for you. After merging this PR into develop, I can also take care of synchronizing #2005 to the updated `develop`.

Thanks, this simplifies my pull request and the Quasi-Newton classes. Looks good to go in my opinion.

**Review by NiklasKotarsky — APPROVED** [review]

**Pull request merged**